repair/repair.cc

```diff
@@ -959,9 +959,22 @@ static future<> repair_cf_range(repair_info& ri,
 // Comparable to RepairSession in Origin
 static future<> repair_range(repair_info& ri, const dht::token_range& range) {
     auto id = utils::UUID_gen::get_time_UUID();
-    return do_with(get_neighbors(ri.db.local(), ri.keyspace, range, ri.data_centers, ri.hosts), [&ri, range, id] (std::vector<gms::inet_address>& neighbors) {
+    repair_neighbors neighbors = ri.neighbors.empty() ? repair_neighbors(get_neighbors(ri.db.local(), ri.keyspace, range, ri.data_centers, ri.hosts)) : ri.neighbors[range];
+    return do_with(std::move(neighbors.all), std::move(neighbors.mandatory), [&ri, range, id] (auto& neighbors, auto& mandatory_neighbors) {
       auto live_neighbors = boost::copy_range<std::vector<gms::inet_address>>(neighbors |
```

Comment:
And then you won't need this do_with(), because they will be provided in ri.

Reply:
This means we have to store all the neighbors for the ranges upfront even if we can calculate on the fly, which is the case for normal repairs.

Reply:
Yes, I agree with providing a function instead of the data.